### PR TITLE
Enable some extra warnings

### PR DESCRIPTION
**Pull request by cgzones**

### CODE REVIEW

File: Affinity.c [hunk 1]
```diff
@@ -64,7 +64,7 @@ Affinity* Affinity_get(const Process* proc, ProcessList* pl) {
          }
       } else {
          unsigned int id;
-         hwloc_bitmap_foreach_begin(id, cpuset);
+         hwloc_bitmap_foreach_begin(id, cpuset)
          Affinity_add(affinity, id);
          hwloc_bitmap_foreach_end();
```
[the remark this answers is not in the document]
Comment (Explorer09) [reply Affinity.c:69]: They should have designed the macro to work like this
```
#define hwloc_bitmap_foreach(id, bitmap, statements) /* implementation detail */
```

[the remark this answers is not in the document]
Comment (Explorer09) [reply Affinity.c:69]: They should have designed the macro to work like this
```
#define hwloc_bitmap_foreach(id, bitmap, statements) /* implementation detail */
```
Shame they didn

[the remark this answers is not in the document]
Comment (Explorer09) [reply Affinity.c:69]: They should have designed the macro to work like this
```
typedef void (*hwloc_bitmap_process_function) (hwloc_bitmap_t bitmap, unsigned int index);
#define hwloc_bitmap_process(bitmap, function) /* implementation detail */
```
It's a shame they didn't.

[the remark this answers is not in the document]
Comment (cgzones) [reply Affinity.c]: Added an extra indentation for the loop body.

File: ColumnsPanel.c [hunk 2]
```diff
@@ -61,6 +61,7 @@ static HandlerResult ColumnsPanel_eventHandler(Panel* super, int ch) {
          }
       }
          /* else fallthrough */
+         ATTR_FALLTHROUGH;
```
[the remark this answers is not in the document]
Comment (BenBE) [reply ColumnsPanel.c:64]: https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wimplicit-fallthrough

I'd prefer `/* FALLTHRU */` (or one of the other comment regexps)

[the remark this answers is not in the document]
Comment (Explorer09) [reply ColumnsPanel.c:64]: https://github.com/hishamhm/htop/pull/664
The fallthrough issue has been discussed already.

File: linux/LinuxProcessList.c [hunk 3]
```diff
@@ -1609,7 +1609,7 @@ static inline void LinuxProcessList_scanMemoryInfo(ProcessList* this) {
                (variable) = parsed_;                                         \
             }                                                                \
             break;                                                           \
-         }
+         } else (void) 0
```
Comment (Explorer09) [inline linux/LinuxProcessList.c:1612]: What's wrong with this? You can't have an else statement after a semicolon.

Comment (cgzones) [reply linux/LinuxProcessList.c:1612]: But one can forget the semicolon:
```c
if (condition)
    tryRead("MemTotal:", totalMem)
else
    printf("Condition not met\n");
```
here the `else` branch will not be associated to `if (condition)` but to the `if` condition inside the macro.

Comment (BenBE) [reply linux/LinuxProcessList.c:1612]: Previously you could write:

```c
tryRead("foo", &foo)
else 
tryRead("bar", &bar)
else
tryRead("baz", &baz)
else {
   // Really not intended …
}
```

This change ensures you have to follow up with a `;` as if this was a normal "function call" …

Comment (Explorer09) [reply linux/LinuxProcessList.c:1612]: @cgzones The `if` already breaks when the condition is true. So concatenating an `else` after this macro won't make sense. And I don't think anyone will use a macro with a dangling else like in your example (it's a control affecting macro, and according to the pattern it's only intended to be used inside a `case` statement).

The only compelling reason to me is to force a semicolon after the macro.
I would add a comment explaining the purpose of the null statement:
```suggestion
         } else (void) 0 /* Require a ";" after the macro use. */
```

**Pull request merged**